server/aesdsocket.c

```diff
@@ -0,0 +1,335 @@
+/*------------------------------------------------------
+ * File Name: aesdsocket.c
+ * Author Name: Abhishek Suryawanshi
+ * Compiler:gcc
+ * linker: g++
+ * Debugger: gdb
+ * References: 
+ * 1. Linux System Programming chapter 5 (Process management) chapter 9(signals)
+ * 2. 
+ -------------------------------------------------------*/
+
+#include <stdarg.h>
+#include <arpa/inet.h>
+#include <signal.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+#include <stdbool.h>
+#include <syslog.h>
+#include <string.h>
+#include <unistd.h>
+#include <stdio.h>
+#include <sys/types.h>
+#include <sys/socket.h>
+#include <netdb.h>
+#include <errno.h>
+#include <stdlib.h>
+#include <arpa/inet.h>
+
+//macros
+#define LOG_FILE_PATH "/var/tmp/aesdsocketdata"
+#define PORT "9000"
+
+//global variable
+int cfd = -1; 
+int sfd = -1;
+struct addrinfo  *updated_addr;
+
+
+
+
+void graceful_exit()
+{
+
+    if(cfd > -1)
+    {
+        shutdown(cfd, SHUT_RDWR);
+        close(cfd);
+    }
+
+    if(sfd > -1)
+    {
+        shutdown(sfd, SHUT_RDWR); 
+        close(sfd);
+    }
+
+    if(updated_addr != NULL)
+    {
+        freeaddrinfo(updated_addr);
+    }
+
+    remove(LOG_FILE_PATH);
```

Comment:
@SuryawanshiAbhishek freeaddrinfo and remove - Are these function safe to do in signal handler?